File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlHint.cs
namespace T1.SqlSharp.Expressions;

public class SqlHint : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.Hint;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_Hint(this);
    }

    public string Name { get; set; } = string.Empty;
    public string ToSql()
    {
        return Name;
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlHint.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlHint implements ISqlExpression {
    SqlType: SqlType = SqlType.Hint;
    Span: TextSpan = new TextSpan();
    Name: string = '';

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_Hint(this);
    }

    ToSql(): string {
        return this.Name;
    }
} 